meeting.js

```diff
@@ -0,0 +1,330 @@
+microphoneButton.addEventListener('click', toggleMicrophone);
+async function toggleMicrophone() {
+  if (!localStream) {
+    console.error('No local stream available.');
+    return;
+  }
+  // Unmute Local Audio
+  if (!isMuted) {
+    try {
+      const microphoneStream = await webex.meetings.mediaHelpers.createMicrophoneStream({
+        echoCancellation: true,
+        noiseSuppression: true,
+      });
+
+      localStream.microphone = microphoneStream;
+      await createdMeeting.publishStreams({ microphone: localStream.microphone });
+      microphoneButton.innerHTML = '<i class="fas fa-microphone icon green"></i>';
+      microphoneButton.style.padding = "14px 21px";
+      isMuted = true;
+    } catch (error) {
+      console.error('Error creating microphone stream:', error);
+    }
+  } 
+  // Mute Local Audio
+  else {
+    if (localStream.microphone) {
+      await createdMeeting.unpublishStreams([localStream.microphone]);
+      localStream.microphone = null;
+      bnrButton.classList.remove('selected');
+    }
+    microphoneButton.innerHTML = '<i class="fas fa-microphone-slash icon red"></i>';
+    microphoneButton.style.padding = "14px 17px";
+    isMuted = false;
+  }
+}
+
+// 16) Added the Local Share System using Share Button.
+shareButton.addEventListener('click', localshare);
+async function localshare() {
+  const [localShareVideoStream, localShareAudioStream] =
+    await webex.meetings.mediaHelpers.createDisplayStreamWithAudio();
+  await createdMeeting.publishStreams({
+    screenShare: {
+      video: localShareVideoStream,
+      audio: localShareAudioStream,
+    }
+  });
+}
+
+// 17) Enabled BNR using Event Listeners and toggleBNR Function.
+bnrButton.addEventListener('click', toggleBNR);
+async function toggleBNR() {
+  let bnrEffect = null;
+  if (!localStream || !localStream.microphone) {
+    console.error('No local microphone stream available.');
+    return;
+  }
+
+  try {
+    if (!bnrEffect) {
+      const audioContext = new AudioContext({ sampleRate: 48000 });
+      bnrEffect = await webex.meetings.createNoiseReductionEffect(audioContext);
+      await localStream.microphone.addEffect(bnrEffect);
+    }
+
+    if (bnrEffect.isEnabled) {
+      await bnrEffect.disable();
+      bnrButton.classList.remove('selected');
+      console.log('BNR disabled');
+    } else {
+      await bnrEffect.enable();
+      console.log('BNR enabled');
+      bnrButton.classList.add('selected');
+    }
+  } catch (error) {
+    console.error('Error toggling BNR:', error);
+  }
+}
+
+// 18) Enabled VBG using Event Listeners and toggleVBG Function.
+vbgButton.addEventListener('click', toggleVBG);
+async function toggleVBG() {
+  if (!localStream || !localStream.camera || !localStream.camera.outputStream) {
+    console.error('No local video stream available.');
+    return;
+  }
+  try {
+    if (!vbgEffect) {
+      vbgEffect = await webex.meetings.createVirtualBackgroundEffect();

```

Comment:
- How does it determine the mode is background blur?
- Can we also support Video Mirroring as part of this? I see that it is possible with the wiki [here](https://github.com/webex/webex-js-sdk/wiki/Streams-and-Effects#virtual-background-effect).

Reply:
You have said to keep background as blur by making Video Mirroring we can keep the background as blur in the documentation given when we not given any parameters in the function the background by default taken as blur we can adjust the blur by adding blur strength.

Reply:
Like I had set mirror default as true if we add check box where to add

Reply:
Instead of having checkbox can we implemented internally through VBG.